### PR TITLE
TreeList: update Column Chooser and Header Filter demos

### PR DESCRIPTION
**Pull request by sergepilipchuk**

### CODE REVIEW

File: JSDemos/Demos/TreeList/ColumnChooser/description.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ Users drag and drop column headers to and from the column chooser.
 * *"select"*         
 Users select and deselect checkboxes with column names.
 
-If the [allowSearch](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/#allowSearch) property is set to **true**, users can search through column names. In this demo, use the checkbox below the TreeList to toggle this feature.
+The column chooser selection is non-recursive. When a user selects a parent item, nested items remain unselected. To select items recursively, set the **selection**.[recursive](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/selection/#recursive) property to **true**. In this demo, you can use the checkbox under the TreeList component to toggle this property.
```
Comment (Abadzhev) [inline JSDemos/Demos/TreeList/ColumnChooser/description.md]: I'm making changes consistent with the DataGrid demos.

```suggestion
In *"select"* mode, you can choose whether parent element selection affects child/nested elements. Use the **selection**.[recursive](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/selection/#recursive) property.
```

File: JSDemos/Demos/TreeList/ColumnChooser/description.md [hunk 2]
```diff
@@ -8,6 +8,8 @@ Users drag and drop column headers to and from the column chooser.
...
+
+If the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/search/#enabled) property is set to **true**, users can search through column names.  In this demo, use the checkbox below the TreeList to toggle this feature.
```
Comment (Abadzhev) [inline JSDemos/Demos/TreeList/ColumnChooser/description.md]:
```suggestion
If the column chooser contains a lot of hidden column, you can enable column search UI. Assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/search/#enabled) property.

In this demo, use checkboxes below the TreeList to toggle the selection and search features.
```

File: JSDemos/Demos/TreeList/UsingHeaderFilter/description.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ Enable the **headerFilter**.[visible](/Documentation/ApiReference/UI_Components/
...
+Users can search values within header filters. To display the search bar, assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/headerFilter/search/#enabled) property in the global [headerFilter](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/headerFilter/) object or in a column's [headerFilter](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/headerFilter/) object. You can also specify a comparison operation used to search header filter values. For this, implement the **search**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/headerFilter/search/#mode) property.
```
Comment (Abadzhev) [inline JSDemos/Demos/TreeList/UsingHeaderFilter/description.md]:
```suggestion
Users can search values within header filters. To display the search bar, assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/headerFilter/search/#enabled) property in the global [headerFilter](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/headerFilter/) object or in a column's [headerFilter](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/headerFilter/) object. You can also specify a comparison operation used to search header filter values. Use the **search**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/headerFilter/search/#mode) property.
```

File: JSDemos/Demos/TreeList/ColumnChooser/description.md [hunk 4]
```diff
@@ -8,6 +8,10 @@ Users drag and drop column headers to and from the column chooser.
 * *"select"*         
 Users select and deselect checkboxes with column names.
```
Comment (albertov05) [inline JSDemos/Demos/TreeList/ColumnChooser/description.md]:
```suggestion
Users select and deselect check boxes with column names.
```

File: JSDemos/Demos/TreeList/ColumnChooser/description.md [hunk 5]
```diff
@@ -8,6 +8,10 @@ Users drag and drop column headers to and from the column chooser.
...
 
-If the [allowSearch](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/#allowSearch) property is set to **true**, users can search through column names. In this demo, use the checkbox below the TreeList to toggle this feature.
+In *"select"* mode, you can choose whether parent element selection affects child/nested elements. Use the **selection**.[recursive](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/selection/#recursive) property.
+
+If the column chooser contains a lot of hidden column, you can enable column search UI. Assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/search/#enabled) property.
```
Comment (albertov05) [inline JSDemos/Demos/TreeList/ColumnChooser/description.md]:
```suggestion
If the column chooser contains many hidden columns, you can enable the column search UI. Assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/search/#enabled) property.
```

File: JSDemos/Demos/TreeList/ColumnChooser/description.md [hunk 6]
```diff
@@ -8,6 +8,10 @@ Users drag and drop column headers to and from the column chooser.
...
+
+In this demo, use checkboxes below the TreeList to toggle the selection and search features.
```
Comment (albertov05) [inline JSDemos/Demos/TreeList/ColumnChooser/description.md]:
```suggestion
In this demo, use the check boxes below the TreeList to toggle the selection and search features.
```

**Review by vladaskorohodova — APPROVED** [review]

**Pull request merged**